### ISSUE TITLE
Support new parameters on Vault aws-ec2 role creation

### ISSUE THREAD

**Issue by ianwestcott** [ianunruh/hvac#108]
Vault's aws-ec2 backend supports multiple login constraints. HVAC knows about the `bound_ami_id` constraint originally introduced with the backend in Vault 0.6.0, but others have since been added: 

- `bound_account_id` (0.6.1)
- `bound_iam_role_arn` (0.6.1)
- `bound_iam_instance_profile_arn` (0.6.2)

HVAC's `create_ec2_role()` method needs to be updated to support the new constraints.

**Comment by ianwestcott** [comment]
This issue is blocking a project I'm working on, so I'm going to attempt to take it on soon. Ideally I'll be able to tackle adding a test suite for EC2 auth as part of this work; I've read the discussion in https://github.com/ianunruh/hvac/issues/59 and plan to explore https://github.com/lyft/metadataproxy as a potential solution. Wish me luck 😄

**Comment by ianwestcott** [comment]
Digging into this a bit more, I think [moto](https://github.com/spulec/moto) is a better fit for the type of mocking we want to do. I'm testing it out now.

**Comment by ianwestcott** [comment]
It turned out that I didn't need to mock anything in order to test role CRUD with Vault, so after getting rabbitholed on moto for a bit I had to rein in the scope of my changes. I have opened a PR for this issue here: https://github.com/ianunruh/hvac/pull/109

In case anyone's curious, I have a separate branch for the moto stuff here: https://github.com/ianunruh/hvac/compare/master...ianwestcott:moto_testing The mock certificates in there were an attempt to write an acceptance test for EC2-based Vault authorization, but I have not yet succeeded in that effort. Suggestions welcome.
